src/mapContainer.ts

```diff
@@ -75,6 +81,63 @@ export type ViewSettings = {
     skipAnimations?: boolean;
 };
 
+type Edge = {
+    v1: leaflet.LatLng;
+    v2: leaflet.LatLng;
+    polyline?: leaflet.Polyline;
+};
+
+type FileWithMarkers = {
+    file: TFile;
+    markers: FileMarker[];
+};
+
+interface FrontMatterAttributes {
+    [key: string]: any; // This allows any property with string keys
+}
+
+class Vertex {
```

Comment:
I may be missing something big & basic here, but I'm unable to understand what the Vertex class is needed for.
Is it merely a logical representation of the map markers, as a separate entity, for the purpose of building the graph before drawing it? Why not just add the notion of edges to the existing `FileMarker` class, build the logical connections between markers as part of the build process (maybe in `finalizeMarkers`?), then update the displayed edges on the map itself?

Reply:
You're not missing anything big and basic. I fully agree with your suggested change. This is an artifact of me still getting familiar with your code and having to pump out a prototype quickly for work. I had plans to get rid of this class but forgot. I will make the changes you suggest above.